### PR TITLE
Delete custom FastRandom from ThreadPool

### PR DESCRIPTION
**Pull request by stephentoub**
- Consolidates 128/256-bit variants of Xoshiro** into the same class name (you're welcome, @danmosemsft  :wink:)
- Uses that from ThreadPool instead of its custom xorshift-based algorithm

Throughput stays the same (~2.5ns per next random value) and incurs just one additional object allocation per thread pool thread.

Follow-up to https://github.com/dotnet/runtime/pull/47085
cc: @kouvel

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems [hunk 1]
```diff
@@ -14,6 +14,7 @@
     <SupportsArmIntrinsics Condition="'$(Platform)' == 'arm64'">true</SupportsArmIntrinsics>
     <SupportsX86Intrinsics Condition="'$(Platform)' == 'x64' or ('$(Platform)' == 'x86' and '$(TargetsUnix)' != 'true')">true</SupportsX86Intrinsics>
     <ILLinkSharedDirectory>$(MSBuildThisFileDirectory)ILLink\</ILLinkSharedDirectory>
+    <Is64Bit Condition="'$(Platform)' == 'arm64' or '$(Platform)' == 'x64'">true</Is64Bit>
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems:17]: Nit: You can also use this to move `DefineConstants` for `TARGET_64BIT` / `TARGET_32BIT` here, and delete them from the runtime-specific files.

**Review by jkotas — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Random.Xoshiro256StarStarImpl.cs [hunk 2]
```diff
@@ -45,6 +45,10 @@ public unsafe Xoshiro256StarStarImpl()
                 while ((_s0 | _s1 | _s2 | _s3) == 0); // at least one value must be non-zero
             }
 
+            /// <summary>Produces a value in the range [0, uint.MaxValue].</summary>
+            [MethodImpl(MethodImplOptions.AggressiveInlining)] // small-ish hot path used by very few call sites
```
Comment (danmosemsft) [inline src/libraries/System.Private.CoreLib/src/System/Random.Xoshiro256StarStarImpl.cs:49]: Surely this one liner inlines itself

Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/Random.Xoshiro256StarStarImpl.cs:49]: I did it for consistency across implementations, so the methods in both cases have the same visibility, same attributes, etc.

File: src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems:17]: Done

File: src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems [hunk 3]
```diff
@@ -14,6 +14,11 @@
...
+  </PropertyGroup>
```
Comment (akoeplinger) [inline src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems:18]: /cc @nealef we'll need to remember to do the same change for s390x once we mirror this to runtimelab.

**Pull request merged**